ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -301,19 +306,26 @@ private void throwOnUnknownColumnReference() {
           new ExpressionAnalyzer(analysis.getFromSourceSchemas());
 
       for (final Expression selectExpression : analysis.getSelectExpressions()) {
-        expressionAnalyzer.analyzeExpression(selectExpression);
+        expressionAnalyzer.analyzeExpression(selectExpression, false);
       }
 
       if (analysis.getWhereExpression() != null) {
-        expressionAnalyzer.analyzeExpression(analysis.getWhereExpression());
+        final boolean allowWindowMetaFields = staticQuery
+            && analysis.getFromDataSources().get(0)
+            .getDataSource()
```

Comment:
Can we be sure there is at least one datasource? Other will NPE. Might be worth a sanity check.

Reply:
There's always at least one source, by definition.  In this case there is a check before to handle joins, so there must be only a single source by this point.